test/qicore410_test.js

```diff
@@ -0,0 +1,138 @@
+    expect(newCurrent).to.deep.equal(nextPatient);
+  });
+
+  it('should clone currentPatient each time it is called', () => {
+    expect(patientSource.currentPatient()).to.deep.equal(patientSource.currentPatient());
+    expect(patientSource.currentPatient()).to.not.equal(patientSource.currentPatient());
+  });
+
+  it('should find patient birthDate', () => {
+    const pt = patientSource.currentPatient();
+    // cql-execution v1.3.2 currently doesn't export the new Date class, so we need to use the .getDate() workaround
+    expect(compact(pt.get('birthDate'))).to.deep.equal({
+      value: new cql.DateTime.parse('1995-01-01').getDate()
+    });
+    expect(pt.get('birthDate.value')).to.deep.equal(new cql.DateTime.parse('1995-01-01').getDate());
+  });
+
+  it('should find patient extensions', () => {
+    const pt = patientSource.currentPatient();
+    const extensions = pt.get('extension');
+    expect(extensions).to.have.length(2);
+    //Check the first and last ones
+    expect(compact(extensions[0])).to.deep.equal({
+      url: { value: 'http://hl7.org/fhir/us/core/StructureDefinition/us-core-race' },
+      extension: [
+        {
+          url: { value: 'ombCategory' },
+          value: {
+            system: { value: 'urn:oid:2.16.840.1.113883.6.238' },
+            code: { value: '2028-9' },
+            display: { value: 'Asian' }
+          }
+        }
+      ]
+    });
+  });
+
+  it('should find records by type name (e.g., Encounter)', () => {
+    const pt = patientSource.currentPatient();
+    const encounters = pt.findRecords('Encounter');
+    expect(encounters).to.have.length(1);
+    expect(encounters.every(c => c.getTypeInfo().name === 'Encounter')).to.be.true;
+    const paymentReconciliations = pt.findRecords('PaymentReconciliation');
+    expect(paymentReconciliations).to.be.empty;
+  });
+
+  it('should find records by model name and type name (e.g., FHIR.Encounter)', () => {
+    const pt = patientSource.currentPatient();
+    const encounters = pt.findRecords('FHIR.Encounter');
+    expect(encounters).to.have.length(1);
+    expect(encounters.every(c => c.getTypeInfo().name === 'Encounter')).to.be.true;
+    const paymentReconciliations = pt.findRecords('FHIR.PaymentReconciliation');
+    expect(paymentReconciliations).to.be.empty;
+  });
+
+  it('should find records by model URL and type name (e.g., {http://hl7.org/fhir}Encounter)', () => {
+    const pt = patientSource.currentPatient();
+    const encounters = pt.findRecords('{http://hl7.org/fhir}Encounter');
+    expect(encounters).to.have.length(1);
+    expect(encounters.every(c => c.getTypeInfo().name === 'Encounter')).to.be.true;
+    const paymentReconciliations = pt.findRecords('{http://hl7.org/fhir}PaymentReconciliation');
+    expect(paymentReconciliations).to.be.empty;
+  });
+
+  it('should find records by QICore profile URL (e.g., http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-encounter)', () => {
```

Comment:
Now that we aren't using the meta.profile fallback, we can rework this test a bit to actually test what we want. Without the `meta.profile` fallback, which we don't actually want in the code, this test as is will fail.

What we should be testing instead is that when `findRecords` is called with with a qicore profile in as the url, AND a second argument that includes the proper datatype, we still get the assertions.

AKA the ask here is to see if adding proper retrieveDetails in the `findRecords` call here makes this test pass